tests/Doctrine/Tests/ORM/Functional/Ticket/GH6823Test.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH6823Test extends OrmFunctionalTestCase
+{
+    public function testCharsetCollationWhenCreatingForeignRelations(): void
+    {
+        if (! $this->_em->getConnection()->getDatabasePlatform() instanceof MySQLPlatform) {
+            self::markTestSkipped('This test is useful for all databases, but designed only for mysql.');
+        }
+
+        $this->createSchemaForModels(
+            GH6823User::class,
+            GH6823Group::class,
+            GH6823Status::class
+        );
+
+        $queries = $this->_sqlLoggerStack->queries;
```

Comment:
This is going to conflict with 328f36846e8f194d02fb7abd3434a8de43859002 when merging up

Reply:
What do you suggest?

Reply:
I see several solutions:
- implement something similar (not sure how feasable), might imply porting 328f368
- contribute this bugfix to 2.12.x. Since you describe this as "being smart", and not "being correct", it might make sense.

Reply:
I think in the end, this is really a bugfix. Even though I call it being smart, the produced SQL by DBAL is just invalid, as it doesn't work in MySQL (InnoDB). 

Is there an ETA for 2.12.0? If it will be released within 4 weeks I can live with that and can just target the 2.12.x branch.

Reply:
Let me ask internally if I can release it, there are a lot of commits to release already.

Reply:
I updated the PR to target 2.12.x. Could you approve the workflow one more time so that we can see it goes green?

Reply:
Somehow the tests starts to fail now. It it possible to allow my workflows without approval? I might need to push more commits to fix this problem, that I cannot reproduce locally.

Reply:
> Somehow the tests starts to fail now. It it possible to allow my workflows without approval? I might need to push more commits to fix this problem, that I cannot reproduce locally.

Yes, for that I would need to merge one of your PRs. Can you find something small to fix, maybe  in the docs or the readme?

Reply:
@greg0ire Done, https://github.com/doctrine/orm/pull/9639